File: Fundamentals_of_Software_Engineering/front/lk/common/src/commonMain/kotlin/org/suai/common/LoginManager.java
package org.suai.common;

public class LoginManager {

    public Response validate(String login, String password, String secondPassword){
        if(!password.equals(secondPassword)){
            return new Response(Role.NONE, "invalid pass");
        }

        return new Response(Role.NONE, "ok");
    }
}

File: Fundamentals_of_Software_Engineering/front/lk/common/src/commonMain/kotlin/org/suai/common/LoginManager.kt
package org.suai.common

//import org.suai.common.model.loginModel.login
import org.suai.common.model.loginModel
import org.suai.common.LoginRepository

class LoginManager {
    /**
     * Проверяет данные на клиенте, затем передает их на сервер и возвращает
     * от него ответ
     * @param loginModel Данные (логин и пароль)
     * @return Response Ответ о правильности данных либо статус операции
     */
    fun sendToServer(loginModel: loginModel): Response {
        return validate(loginModel)
    }

    private fun validate(data: loginModel): Response {
        return if (data.login == "") {
            Response("Login is empty")
        } else sendToRepository(data)
    }

    private fun sendToRepository(data: loginModel): Response {
        return LoginRepository.getResponse(data)
    }
}